src/main/java/com/snowplowanalytics/snowplow/tracker/emitter/BatchEmitter.java

```diff
@@ -88,13 +94,16 @@ protected BatchEmitter(final Builder<?> builder) {
         Preconditions.checkArgument(builder.bufferSize > 0, "bufferSize must be greater than 0");
 
         this.bufferSize = builder.bufferSize;
-        this.eventStore = builder.eventStore;
+        this.bufferCapacity = builder.bufferCapacity;
+//        this.eventStore = builder.eventStore;
 
-        checkForEventsToSend = new Thread(
-                getCheckForEventsToSendRunnable(),
-                EVENTS_CHECK_THREAD_NAME_PREFIX + EVENTS_CHECK_THREAD_NUMBER.getAndIncrement()
-        );
-        checkForEventsToSend.start();
+        if (builder.eventStore == null) {
+            this.eventStore = new InMemoryEventStore(this.bufferCapacity);
+        } else {
+            this.eventStore = builder.eventStore;
+        }
+
+        this.retryDelay = new AtomicLong(0);
```

Comment:
Opinion about code style: Definitely not a practice in this codebase but I think we should avoid the use of the `this` keyword if there is a problem of ambiguity.
A Linter can help but not a problem now. Nothing to change now btw, it's just a suggestion for the future.